test/test_basic_rng/rng_common.hh

```diff
@@ -195,3 +210,53 @@ double hypergeometric_variance(int64_t N, int64_t K, int64_t D) {
 }
 
 } // end namespace RandBLAS_StatTests
+
+
+
+//
+// MARK Kolmogorov-Smirnov Calculations
+//
+
+// Function to compute the K-S statistic
+double ks_stat(const std::vector<double> &cdf1, const std::vector<double> &cdf2)
+{
+    assert(cdf1.size() == cdf2.size()); // Vectors must be of same size to perform test
+
+    // Loop through and keep track of max
+    double max_diff = 0.0;
+    for (size_t i = 0; i < cdf1.size(); ++i)
+    {
+        double diff = std::abs(cdf1[i] - cdf2[i]);
+        if (diff > max_diff)
+        {
+            max_diff = diff;
+        }
+    }
+    return max_diff;
+}
```

Comment:
It's preferable for us to know the first index where the test would fail. One way forward is to change this function to accept a tolerance and return a ``std::tuple<int, double>`` of the form ``{i, first_violation}`` where ``i`` is the first index where ``first_violation = std::abs(cdf1[i] - cdf2[i])`` exceeds the tolerance.

Reply:
Hmm, ok. This is not so much what I expected. Is the intuition here that sharing '0' elements should be very close to real randomness, whereas sharing 'k' elements will almost surely not represent real randomness? Meaning, the goal of the function is to tell when this sort of switch from random to not random happens? (according to some threshold)

Reply:
My initial comment probably wasn't clear. I'm just suggesting that you modify the function as follows
```c++
std::pair<int,double> check_ks_below_critval(const std::vector<double> &cdf1, const std::vector<double> &cdf2, double critical_value)
{
    assert(cdf1.size() == cdf2.size()); // Vectors must be of same size to perform test

    for (size_t i = 0; i < cdf1.size(); ++i) {
        double diff = std::abs(cdf1[i] - cdf2[i]);
        if (diff > critical_value) {
            return {i, diff}; // the test failed.
        }
    }
    return {-1, 0.0};  // interpret a negative return value as the test passing.
}
```
Presumably you were computing ``ks_stat`` and then checking if it exceeded a given critical value. I'm suggesting that you do both steps in one function. This is useful because if the test fails then we immediately can get information that's useful for debugging (the index i where the empirical and true CDFs differ by more than the critical value).

Does that make sense?

Reply:
I see, this does make sense. Yep, you have the right assessment of what I was doing before. I was approaching it from more of a "one-time test" where we could see a grid of significance values, but see now that's it's more of a debugging process.